RetailCoder.VBE/Inspections/QuickFixes/AssignedByValParameterMakeLocalCopyQuickFix.cs

```diff
@@ -118,74 +114,56 @@ private string BuildLocalCopyAssignment()
                 + _localCopyVariableName + " = " + _target.IdentifierName;
         }
 
-        private IEnumerable<string> GetIdentifierNamesAccessibleToProcedureContext(RuleContext ruleContext)
+        private IEnumerable<string> GetIdentifierNamesAccessibleToProcedureContext()
         {
-            var allIdentifiers = new HashSet<string>();
-
-            var blocks = QuickFixHelper.GetBlockStmtContextsForContext(ruleContext);
-
-            var blockStmtIdentifiers = GetIdentifierNames(blocks);
-            allIdentifiers.UnionWith(blockStmtIdentifiers);
-
-            var args = QuickFixHelper.GetArgContextsForContext(ruleContext);
-
-            var potentiallyUnreferencedParameters = GetIdentifierNames(args);
-            allIdentifiers.UnionWith(potentiallyUnreferencedParameters);
-
-            //TODO: add module and global scope variableNames to the list.
-
-            return allIdentifiers.ToArray();
+            var allSameProcedureDeclarations = _parserState.AllUserDeclarations
```

Comment:
I would avoid fetching `AllUserDeclarations` more than once; the `Where` predicate could encompass everything that's needed in one pass - but looking at the predicates below perhaps each should be in its own `bool` function. As a result you wouldn't need to populate a list, you could just return the (distinct) identifier names directly.